projects/packages/my-jetpack/src/class-rest-products.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Sets up the Products REST API endpoints.
+ *
+ * @package automattic/my-jetpack
+ */
+
+namespace Automattic\Jetpack\My_Jetpack;
+
+require_once 'class-products.php';
+
+/**
+ * Registers the REST routes for Products.
+ */
+class REST_Products {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		register_rest_route(
+			'my-jetpack/v1',
+			'/site/products',
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => __CLASS__ . '::get_products',
+				'permission_callback' => __CLASS__ . '::permissions_callback',
+			)
+		);
+
+		register_rest_route(
+			'my-jetpack/v1/',
+			'/site/products/(?P<product>[a-z\-]+)',
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => __CLASS__ . '::get_product',
```

Comment:
> This callback is missing, just checking if it was intentional. 

It wasn't but a mistake produced by working in two branches at the same time :clap:. Addressed here 7d2417c

> Also, I'd say that the check to see if the product exist and return 404 could be in this missing callback, and not in the permission check callback

I moved the method to the permission one because we are going to reuse it for a `[POST]` method.

Reply:
Actually, checking endpoint args there is not the best option. Updating...

Reply:
Addressed here: https://github.com/Automattic/jetpack/pull/22389/commits/f6329082adc6a0a40efba53e808d4504abfb1de2. Setting as resolved.